File: app/downloaders/_go_files/json_process/json_process.py
import ijson, orjson, asyncio, time

async def _test() -> None:

	_book_title = ''
	_book_url = ''
	_author_name = ''
	_author_url = ''
	_seria_name = ''
	_seria_number = ''
	_seria_url = ''
	_chapters = []
	_chapter = None
	with open("t2.json", "rb") as f:
		parser = ijson.parse(f)
		for prefix, event, value in parser:
			if prefix == 'Title':
				_book_title = value
			if prefix == 'Url':
				_book_url = value
			if prefix == 'Author.Name':
				_author_name = value
			if prefix == 'Author.Url':
				_author_url = value
			if prefix == 'Seria.Name':
				_seria_name = value
			if prefix == 'Seria.Number':
				_seria_number = value
			if prefix == 'Seria.Url':
				_seria_url = value
			if prefix == 'Chapters.item' and event == 'start_map':
				_chapter = {}
			if prefix == 'Chapters.item.Title':
				_chapter['Title'] = value
			if prefix == 'Chapters.item.IsValid':
				_chapter['IsValid'] = value
			if prefix == 'Chapters.item' and event == 'end_map':
				_chapters.append(_chapter)
				_chapter = None

	_json = orjson.dumps({
		"book_title":   _book_title,
		"book_url":     _book_url,
		"author_name":  _author_name,
		"author_url":   _author_url,
		"seria_name":   _seria_name,
		"seria_number": _seria_number,
		"seria_url":    _seria_url,
		"chapters":     _chapters,
	})

	with open("res_py.json", "wb") as f:
		f.write(_json)

start_time = time.time()
asyncio.run(_test())
elapsed = time.time() - start_time
print(f"Execution time: {elapsed}s")

File: app/downloaders/_go_files/json_process/json_process.go
package main

import (
	"encoding/json"
	"github.com/buger/jsonparser"
	"os"
)

var _book_title string = ""
var _book_url string = ""
var _author_name string = ""
var _author_url string = ""
var _seria_name string = ""
var _seria_number string = ""
var _seria_url string = ""
var _chapters = []_chapter{}

type _chapter struct {
	Title   string `json:"title"`
	IsValid bool   `json:"valid"`
}

type Result struct {
	Book_title   string     `json:"book_title"`
	Book_url     string     `json:"book_url"`
	Author_name  string     `json:"author_name"`
	Author_url   string     `json:"author_url"`
	Seria_name   string     `json:"seria_name"`
	Seria_number string     `json:"seria_number"`
	Seria_url    string     `json:"seria_url"`
	Chapters     []_chapter `json:"chapters"`
}

func parseOne(key []byte, value []byte, dataType jsonparser.ValueType, offset int) error {

	_key := string(key)

	if _key == "Title" {
		_book_title, _ = jsonparser.ParseString(value)
	}
	if _key == "Url" {
		_book_url, _ = jsonparser.ParseString(value)
	}
	if _key == "Author" {
		_author_name, _ = jsonparser.GetString(value, "Name")
		_author_url, _ = jsonparser.GetString(value, "Url")
	}
	if _key == "Seria" {
		_seria_name, _ = jsonparser.GetString(value, "Name")
		_seria_number, _ = jsonparser.GetString(value, "Number")
		_seria_url, _ = jsonparser.GetString(value, "Url")
	}
	if _key == "Chapters" {
		jsonparser.ArrayEach(value, func(item []byte, itemDataType jsonparser.ValueType, itemOffset int, err error) {
			_Title, _ := jsonparser.GetString(item, "Title")
			_IsValid, _ := jsonparser.GetBoolean(item, "IsValid")

			chapter := _chapter{Title: _Title, IsValid: _IsValid}
			_chapters = append(_chapters, chapter)
		})
	}
	return nil
}

func main() {
	orig := os.Args[1]

	fileName := orig
	data, err := os.ReadFile(fileName)
	if err != nil {
		panic(err)
	}
	jsonparser.ObjectEach(data, parseOne)

	res := Result{Book_title: _book_title, Book_url: _book_url, Author_name: _author_name, Author_url: _author_url, Seria_name: _seria_name, Seria_number: _seria_number, Seria_url: _seria_url, Chapters: _chapters}

	file, err1 := json.Marshal(res)
	if err1 != nil {
		panic(err1)
	}

	err2 := os.WriteFile(orig, file, 0644)
	if err2 != nil {
		panic(err2)
	}
}
